tests/Silverback.Integration.Tests.E2E/Kafka/CallbacksTests.cs

```diff
@@ -151,6 +156,177 @@ public async Task EndpointsConfiguredCallback_MessagePublished()
+        [InlineData(false, 0)]
+        public async Task EndpointsConfiguredKafkaEndOfTopicPartitionReachedCallback(bool enablePartitionEof, int expectedEndOfPartitionReachedCount)
+        {
+            var message = new TestEventOne
+            {
+                Content = "Hello E2E!"
+            };
+
+            Host.ConfigureServices(
+                    services => services
+                        .AddLogging()
+                        .AddSilverback()
+                        .UseModel()
+                        .AddSingletonBrokerCallbackHandler<KafkaEndOfTopicPartitionReachedCallback>()
+                        .WithConnectionToMessageBroker(
+                            options => options
+                                .AddMockedKafka())
+                        .AddKafkaEndpoints(
+                            endpoints => endpoints
+                                .Configure(
+                                    config =>
+                                    {
+                                        config.BootstrapServers = "PLAINTEXT://tests";
+                                    })
+                                .AddOutbound<IIntegrationEvent>(
+                                    endpoint => endpoint.ProduceTo(DefaultTopicName))
+                                .AddInbound(
+                                    endpoint => endpoint
+                                        .ConsumeFrom(DefaultTopicName)
+                                        .Configure(
+                                            config =>
+                                            {
+                                                config.GroupId = "consumer1";
+                                                config.EnablePartitionEof = enablePartitionEof;
+                                            })))
+                        .AddIntegrationSpyAndSubscriber())
+                .Run();
+
+            var publisher = Host.ScopedServiceProvider.GetRequiredService<IEventPublisher>();
+            await publisher.PublishAsync(message);
+
+            var callbackHandlers = Host.ScopedServiceProvider
+                .GetServices<IBrokerCallback>()
+                .Where(service => service is KafkaEndOfTopicPartitionReachedCallback)
+                .ToList();
+
+            var kafkaTestingHelper = Host.ServiceProvider.GetRequiredService<IKafkaTestingHelper>();
+            await kafkaTestingHelper.WaitUntilAllMessagesAreConsumedAsync();
+
+            callbackHandlers.Should().HaveCount(1);
+            callbackHandlers.Cast<KafkaEndOfTopicPartitionReachedCallback>().Should().OnlyContain(handler => handler.EndOfPartitionReachedCount == expectedEndOfPartitionReachedCount);
+
+            kafkaTestingHelper.Spy.OutboundEnvelopes.Should().HaveCount(1);
+        }
+
+        [Fact]
+        public async Task EndpointsConfiguredKafkaEndOfTopicPartitionReachedCallback_WithProducingSeveralMessages()
```

Comment:
Method name doesn't follow the in place convention.

Reply:
Isn't worth testing a case where you publish to only one of the two partitions and assert that only one EOF is received?